monai/bundle/scripts.py

```diff
@@ -116,6 +121,212 @@ def _get_fake_spatial_shape(shape: Sequence[Union[str, int]], p: int = 1, n: int
+
+        # Execute this module as a CLI entry, and download the whole bundle package:
+        python -m monai.bundle download --package <bundle package name> --source "github" --repo <repo name>
+
+        # Execute this module as a CLI entry, and download a single file:
+        python -m monai.bundle download --package <bundle package name> --filename <file name> --repo <repo name>
+
+        # Set default args of `run` in a JSON / YAML file, help to record and simplify the command line.
+        # Other args still can override the default args at runtime:
+        python -m monai.bundle download --args_file "/workspace/data/args.json" --filename <file name>
+
+    Args:
+        package: the bundle package name. If `None` and `url` is `None`, it must be provided in `args_file`.
+            If `source` is `github`, it should be the same as the release tag.
+        bundle_dir: target directory to store the download data. Default is `bundle` subfolder under`torch.hub get_dir()`.
+        filename: the filename that needs to be downloaded.
+            If `source` is `github` and filename is `None`, the whole bundle package will be downloaded.
+        source: the place that saved the bundle package.
+            If `source` is `github`, the bundle package should be within the releases.
+        repo: the repo name. If `None` and `url` is `None`, it must be provided in `args_file`.
+            If `source` is `github`, it should be in the form of `repo_owner/repo_name`.
+            For example: `Project-MONAI/MONAI`.
+        url: the url to download the data. It is an optional argument and if not `None`,
+            data will be downloaded directly and `source` will not be checked.
+        progress: whether to display a progress bar.
+        extract: whether to extract the downloaded file. This argument only works when download a single file.
+        args_file: a JSON or YAML file to provide default values for all the args in this function.
+            so that the command line inputs can be simplified.
+
+    """
+    _args = _update_args(
+        args=args_file,
+        package=package,
+        bundle_dir=bundle_dir,
+        filename=filename,
+        source=source,
+        repo=repo,
+        url=url,
+        progress=progress,
+        extract=extract,
+    )
+    if "package" not in _args and url is None:
+        raise ValueError(f"To download from source: {source}, `package` must be provided.")
+    if "repo" not in _args and url is None:
+        raise ValueError(f"To download from source: {source}, `repo` must be provided.")
+    _log_input_summary(tag="download", args=_args)
+    package_, bundle_dir_, filename_, source_, repo_, url_, progress_, extract_ = _pop_args(
+        _args, package=None, bundle_dir=None, filename=None, source="github", repo=None, url=None, progress=True, extract=False
+    )
+
+    if not bundle_dir_:
+        get_dir, has_home = optional_import("torch.hub", name="get_dir")
```

Comment:
I think no need to consider the case that no `bundle_dir` provided so far.

Thanks.

Reply:
Thanks, I think it is helpful in `def load`, but no need to put in `def download`